ebu_tt_live/carriage/filesystem.py

```diff
@@ -115,44 +133,75 @@ def resume_producing(self):
+                counter=sequence_number, 
+                sequence_identifier=sequence_identifier)
+            
+        # TODO: consider using different classes or functions to do the document writing,
+        # depending on the settings of suppress_manifest and rotating_buf etc that
+        # can be selected once at the beginning and dereferenced rather than repeating
+        # if statements.
         filepath = os.path.join(self._dirpath, filename)
-        with open(filepath, 'w') as f:
-            f.write(data)
-        time_base = time_base
-        availability_time = availability_time
-        if delay is not None:
-            availability_time += timedelta(seconds=delay)
-        new_manifest_line = '{},{}\n'.format(timedelta_to_str_manifest(availability_time), filename)
-        self._manifest_content += new_manifest_line
-        with open(self._manifest_path, 'a') as f:
-            f.write(new_manifest_line)
+        with open(filepath, 'w') as destfile:
+            destfile.write(data)
+            destfile.flush()
+                        
+        # If we're running a rotating buffer remember this file for possible deletion later
+        if self._circular_buf_size > 0 :
+            self._circular_buf.append(filepath)
+            
+        if not self._suppress_manifest :
+            # Work out what time to put in the manifest file
+            
+            # NOTE: This is nasty
+            availability_time = self.check_availability_time(
+                sequence_identifier=sequence_identifier,
+                time_base=time_base,
+                clock_mode=clock_mode,
+                availability_time=availability_time
+            )
+
+            if availability_time is None:
+                # This is a possibility with a live messages as first document.
+                # They don't contain enough timing info.
+                log.warning(
+                    FS_MISSING_AVAILABILITY.format(
+                        sequence_identifier=sequence_identifier,
+                        file_path=filepath)
+                )
+                # Without availability time we can not create manifest file.
+                # In this case we have written an output file but no matching
+                # entry in the manifest file.
+                return
+
+            time_base = time_base
+            availability_time = availability_time
+            if delay is not None:
+                availability_time += timedelta(seconds=delay)
+
+            # Open the manifest filepath
+            if self._manifest_path is None :
+                manifest_filename = "manifest_" + sequence_identifier + ".txt"
+                self._manifest_path = os.path.join(self._dirpath, manifest_filename)
+                
+            # Write a new line to the manifest file
+            new_manifest_line = '{},{}\n'.format(timedelta_to_str_manifest(availability_time), filename)
+            self._manifest_content += new_manifest_line
```

Comment:
Since we don't do anything with manifest_content, I don't know why we bother doing this.